web/api/v2/api.go

```diff
@@ -227,3 +250,64 @@ func (s *Admin) DeleteSeries(_ old_ctx.Context, r *pb.SeriesDeleteRequest) (*pb.
 	}
 	return &pb.SeriesDeleteResponse{}, nil
 }
+
+// RemoteWrite receives a stream of write requests and performs a remote write action with them
+func (s *Admin) RemoteWrite(stream pb.Admin_RemoteWriteServer) error {
+	for {
+		resp, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			level.Error(s.logger).Log("msg", "read from grpc stream failure", "err", err)
+			return err
+		}
+
+		// Write all metrics sent
+		WriteTimeSeries(resp.GetTimeseries(), s.db, s.logger)
+	}
+	return nil
+}
+
+// WriteTimeSeries writes a set of timeseries metrics to the tsdb
+func WriteTimeSeries(timeseries []pb.TimeSeries, tsdb func() *tsdb.DB, logger log.Logger) {
+	ap := tsdb().Appender()
+
+	commit := func() {
+		if err := ap.Commit(); err != nil {
+			level.Error(logger).Log("msg", "failure trying to commit write to store", "err", err)
+			if err := ap.Rollback(); err != nil {
+				level.Error(logger).Log("msg", "failure trying to rollback write to store", "err", err)
+			}
+		}
+	}
+	defer commit()
+
+	for i, ts := range timeseries {
+		if i%500 == 0 {
```

Comment:
probably be good to put back the commitChunkSize constant

Reply:
u rite